internal/services/network/subnet_resource_test.go

```diff
@@ -735,6 +753,26 @@ resource "azurerm_subnet" "test" {
 `, r.template(data))
 }
 
+func (r SubnetResource) defaultOutbound(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+resource "azurerm_subnet" "internal" {
+  name                            = "internal"
+  resource_group_name             = azurerm_resource_group.test.name
+  virtual_network_name            = azurerm_virtual_network.test.name
+  address_prefixes                = ["10.0.2.0/24"]
+  default_outbound_access_enabled = false
+}
+resource "azurerm_subnet" "public" {
+  name                            = "public"
+  resource_group_name             = azurerm_resource_group.test.name
+  virtual_network_name            = azurerm_virtual_network.test.name
+  address_prefixes                = ["10.0.3.0/24"]
+  default_outbound_access_enabled = true
+}
+`, r.template(data))
+}
```

Comment:
Which'd require us to make this:

```suggestion
func (r SubnetResource) defaultOutbound(data acceptance.TestData, defaultOutboundAccessEnabled bool) string {
	return fmt.Sprintf(`
%s
resource "azurerm_subnet" "test" {
  name                            = "internal"
  resource_group_name             = azurerm_resource_group.test.name
  virtual_network_name            = azurerm_virtual_network.test.name
  address_prefixes                = ["10.0.2.0/24"]
  default_outbound_access_enabled = %t
}
`, r.template(data), defaultOutboundAccessEnabled)
}
```